sys/net/application_layer/gcoap/forward_proxy.c

```diff
@@ -525,13 +529,20 @@ static uint8_t _cep_get_req_etag_len(client_ep_t *cep)
     return 0;
 }
 
-static void _cep_set_req_etag_len(client_ep_t *cep, uint8_t req_etag_len)
+static void _cep_set_req_etag(client_ep_t *cep, const void *etag,
+                              uint8_t etag_len)
 {
-    if (IS_USED(MODULE_NANOCOAP_CACHE)) {
+    (void)cep;
+    (void)etag;
+    (void)etag_len;
+#if MODULE_NANOCOAP_CACHE
+    if (etag_len <= COAP_ETAG_LENGTH_MAX) {
```

Comment:
```suggestion
    if (IS_USED(MODULE_NANOCOAP_CACHE) && (etag_len <= COAP_ETAG_LENGTH_MAX)) {
```

Alternatively, if you insist on pre-processor magic,

```suggestion
#if IS_USED(MODULE_NANOCOAP_CACHE)
    if (etag_len <= COAP_ETAG_LENGTH_MAX) {
```

? Then you can remove the `(void)` casts above.

Reply:
One of the `struct` members is only available for with `MODULE_NANOCOAP_CACHE`, so no way around using the preprocessor check.

One could indeed do a C check and nest a preprocessor check inside, but that increases the complexity of the code.